### PR TITLE
#142 Store StudentImage in database

### PR DESCRIPTION
**Pull request by schaemik**
- Storing image on tablet
- Create a database entry for every studentimage

### CODE REVIEW

File: app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java [hunk 1]
```diff
@@ -154,12 +160,13 @@ private void storeStudentImage(Mat img){
 
         MatName matName = new MatName(sId, img);
         FileHelper fh = new FileHelper();
-        String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + collectionEventId;
+        String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + sId;
         new File(wholeFolderPath).mkdirs();
         fh.saveMatToImage(matName, wholeFolderPath + "/");
 
         Long Id =  Long.parseLong(String.valueOf((int) (Math.random() * 1000000)));
         StudentImage studentImage = new StudentImage(Id, null, wholeFolderPath, null, null);
```
Comment (sladomic) [inline app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: why 3 x null?

**Review by sladomic — CHANGES_REQUESTED** [review]

**Review by jo-elimu — APPROVED** [review]

File: app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java [hunk 2]
```diff
@@ -154,12 +160,13 @@ private void storeStudentImage(Mat img){
 
         MatName matName = new MatName(sId, img);
         FileHelper fh = new FileHelper();
-        String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + collectionEventId;
+        String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + sId;
```
Comment (jo-elimu) [inline app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: Can we use another name than `sId` to make it more clear what kind of id it is?

File: app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java (hunk 1 above)
Comment (schaemik) [reply app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: - Haven't been sure about the DeviceID yet but will convert it to Hex now and save it in a long.
- @jogrimst how should we store the time? With help of the CalendarConverter? Do you might have an example? 
- Student Image Feature is pending

File: app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java (hunk 2 above)
Comment (schaemik) [reply app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: Yes will change it studenImageId

File: app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java (hunk 1 above)
Comment (jo-elimu) [reply app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: > Haven't been sure about the DeviceID yet but will convert it to Hex now and save it in a long.

@schaemik `deviceId` should look something like "dc53884fdab9ecaf". See `DeviceInfoHelper#getDeviceId`. It returns a String.

Comment (jo-elimu) [reply app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: > @jogrimst how should we store the time? With help of the CalendarConverter? Do you might have an example?

@schaemik Does it work to just put `Calendar.getInstance()` as the value?

Comment (schaemik) [reply app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: @jogrimst Sorry I'm wrong I thought I can save the DeviceID to long but it turns out the DeviceID is 20 digits long and the Long only takes 19. So we have to change it to String on the DB.

Comment (jo-elimu) [reply app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java]: @schaemik Note that the device id can contain letters as well. So it cannot be stored as Long.

**Review by sladomic — APPROVED** [review]

**Pull request merged**